syntax = "proto3";

package xla.gpu;

import "xla/service/buffer_assignment.proto";

// Dimensions of the convolution filter.
// See stream_executor::dnn::FilterDescriptor for more details.
message ConvolutionFilterDimensions {
  // Number of output features (a.k.a. channels).
  int64 output_feature_map_count = 1;
  // Number of input features (a.k.a. channels).
  int64 input_feature_map_count = 2;
  // Height of the filter.
  int64 input_filter_height = 3;
  // Width of the filter.
  int64 input_filter_width = 4;
}

// Buffers for the bias input and output of the convolution reorder thunk.
// Serialized version of xla::gpu::ConvolutionReorderThunk::BiasBuffers.
message ConvolutionReorderBiasBuffers {
  xla.buffer_assignment.BufferAllocationSliceProto bias_input = 1;
  xla.buffer_assignment.BufferAllocationSliceProto bias_output = 2;
}
